userland/Makefile

```diff
@@ -61,7 +61,7 @@ $(BUILDDIR):
 
 $(BUILDDIR)/app.elf: $(OBJS) $(TOCK_USERLAND_BASE_DIR)/newlib/libc.a $(LIBTOCK) | $(BUILDDIR)
 	$(TRACE_LD)
-	$(Q)$(LD) --gc-sections --emit-relocs --entry=_start $(LDFLAGS) -nostdlib $(OBJS) --start-group $(TOCK_USERLAND_BASE_DIR)/newlib/libc.a $(LIBTOCK) --end-group -o $@
+	$(Q)$(CC) -Wl,--gc-sections -Wl,--emit-relocs --entry=_start $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -nostdlib $(OBJS) -Wl,--start-group $(TOCK_USERLAND_BASE_DIR)/newlib/libc.a $(LIBTOCK) -lm -lgcc -Wl,--end-group -o $@
```

Comment:
OK, well, for now, there is actually no way to override `LDFLAGS` outside the Makefile, and all it does is `-T $(LINKER)`. So, how about I just get rid of it and put in `-T $(LINKER)` explicitly?

If anyone ever wants to add more linker arguments we can cross that bridge